web/ui_utils.js

```diff
@@ -985,6 +985,32 @@ function moveToEndOfArray(arr, condition) {
   }
 }
 
+/**
+ * Get the active or focused element in current DOM.
+ *
+ * Recursively search for the truly active or focused element incase there are
+ * shadow DOMs.
+ *
+ * @author Chen Jinlong <chenjinlong2016@outlook.com>
+ *
+ * @param {DocumentOrShadowRoot} root the root element.
```

Comment:
`@param {DocumentOrShadowRoot} [root] - The root element.`

`root` should be in brackets to indicate that it's an optional argument.

We can also leave this out since no code currently calls this function with an argument?